EVM and High Level Languages.md

```diff
@@ -0,0 +1,95 @@
+* Smart contract interaction only by address
+* Arguments and data pass in from the user and pass out to a user or contract
+* A stack type data structure holds local variables
+* It reads function call arguments and return-addresses
+* Logic functions can change the order in which instructions execute
+* Addressable persistent storage acts as a virtual hard drive 
+* Non-persistent storage acts as system memory
+
+A closer look at the EVM and dApp ecosystem show a complex orchestration of silently running programs and processes. Technology today offers many choices in a programming language presenting obstacles in compatibility. Cross-platform data abstraction requires a complex web of behind the scenes computing. 
+
+## Discrete Parts that make the EVM
+
+As stated before, the EVM is prebuilt with libraries of ready to use functions. Smart contract languages such as Solidity and Vyper are simple, feature-rich languages. Embedded function libraries manipulate data in ways making the EVM quasi-Turing complete. Turing complete system is so named on the basis that it can use a lot of math proving it can solve any algorithm.
+
+### EVM Opcodes and Bytecode
+
+<a href="https://imgur.com/J3lKgEy"><img src="https://i.imgur.com/J3lKgEy.png" title="source: imgur.com" /></a>
+
+Opcodes are a large library of functions preprogrammed into the EVM. A list of all Opcodes available is accessible [here](https://www.ethervm.io/#opcodes). Fewer resources handle these functions because they are small, 4-bit hex identifiers. Every function found in JavaScript or Python exists also in the EVM. Opcodes include functions for data manipulation, I/O, storage, security, and logic. High-level smart contract language compiles into bytecode. This is data passed into the EVM in hex that calls opcode functions and handles user arguments. The image located at the top of the page is a matrix of two digit hex opcodes. 
+
+### Transactions
+
+Smart Contracts are simple to deploy on the EVM blockchain. Every type of transaction uses the same format to execute the EVM. Two main smart contact transactions call the contract or deploy a new contract. Either of these transactions are capable of calling a large variety of functions. Functions called can be user defined. Data passes in and out of the EVM in either case. This data can pass back to the user or forward on to another contract.
+
+<a href="https://imgur.com/bfKk7mY"><img src="https://i.imgur.com/bfKk7mY.png" title="source: imgur.com" /></a>
+
+The image above shows how a transaction flow can happen in the EVM. Data is passed out of the EVM in what is known as a 'block'. A block is the final result of the arguments, variables, functions, and transactions presented. The data has been manipulated and encrypted. A proof of work puzzle solved by network miners cryptographically proves data integrity. It also cryptographically ties the completed block to adjacent data blocks.
```

Comment:
```suggestion
The image above shows how a transaction flow happens in the EVM. Data is passed out of the EVM in what is known as a 'block'. A block is the final result of the arguments, variables, functions, and transactions presented. The data has been manipulated and encrypted. A proof of work puzzle solved by network miners cryptographically proves data integrity. It also cryptographically ties the completed block to adjacent data blocks.
```

Reply:
What do you mean by "manipulated"? What's proof of work? Few details missing here